config/software/libffi.rb

```diff
@@ -16,32 +15,52 @@
 #
 
 name "libffi"
-default_version "3.0.13"
 
-dependency "libgcc"
-dependency "libtool"
+default_version "3.2.1"
 
-source :url => "ftp://sourceware.org/pub/libffi/libffi-3.0.13.tar.gz",
-       :md5 => "45f3b6dbc9ee7c7dfbbbc5feba571529",
-       :extract => :seven_zip
+license "MIT"
+license_file "LICENSE"
+skip_transitive_dependency_licensing true
 
-relative_path "libffi-3.0.13"
+# Is libtool actually necessary? Doesn't configure generate one?
+dependency "libtool" unless windows?
 
-env = with_embedded_path()
-env = with_standard_compiler_flags(env)
+version("3.0.13") { source md5: "45f3b6dbc9ee7c7dfbbbc5feba571529" }
+version("3.2.1")  { source md5: "83b89587607e3eb65c70d361f13bab43" }
+
+source url: "ftp://sourceware.org/pub/libffi/libffi-#{version}.tar.gz"
+
+relative_path "libffi-#{version}"
 
 build do
-  ship_license "https://raw.githubusercontent.com/atgreen/libffi/master/LICENSE"
```

Comment:
let's use `ship_license` everywhere instead of `license`/`license_file` until we move all our software def to these newer DSL methods